macrosynergy/panel/view_timelines.py

```diff
@@ -147,195 +127,100 @@ def view_timelines(
-        max_plots = len(xcats)
-    elif (len(cids) > 1) and not single_chart:
-        max_plots = len(cids)
-    ncol: int = min(ncol, max_plots)
-
-    sns.set(style="darkgrid")
-    cross_mean: Optional[pd.DataFrame] = None
-    cs_label: str = f"cross-sectional average of {xcats[0]}."
-    if cs_mean and (len(cids) > 1):
-        dfw: pd.DataFrame = df.pivot(index="real_date", columns="cid", values="value")
-        cm_series: pd.Series = dfw.mean(axis=1)
-        cross_mean: pd.DataFrame = pd.DataFrame(
-            data=cm_series.to_numpy(), index=cm_series.index, columns=["average"]
-        ).reset_index(level=0)
+    
+    face_plotter: FacetPlot = FacetPlot(
+        df=df, cids=cids, xcats=xcats, metrics=[val], start_date=start, end_date=end
+    )
+    line_plotter: LinePlot = LinePlot(
+        df=df, cids=cids, xcats=xcats, metrics=[val], start_date=start, end_date=end
+    )
 
+    cross_mean: pd.Series = None
+    cross_mean_label: str = None
+    if cs_mean:
+        cross_mean_label: str = f"cross-sectional average of {xcats[0]}"
         if xcat_labels is not None:
-            if len(xcat_labels) == (len(xcats) + 1):
-                cs_label = xcat_labels.pop(-1)
-
-    # Replace xcats in the DF with their corresponding labels
-    if xcat_labels is not None:
-        for xc, xl in zip(xcats, xcat_labels):
-            df["xcat"] = df["xcat"].replace(xc, xl)
-        xcats = xcat_labels
-
-    sns.set(rc={"figure.figsize": size})
-    plt.rcParams["figure.figsize"] = size
-
-    fg: Optional[sns.FacetGrid] = None
-    ax: Optional[plt.Axes] = None
-    if len(cids) == 1:
-        if xcat_grid:
-            fg: sns.FacetGrid = sns.FacetGrid(
-                df,
-                col="xcat",
-                col_wrap=ncol,
-                sharey=same_y,
-                height=height,
-                aspect=aspect,
-                col_order=xcats,
-            )
-            fg.map_dataframe(
-                sns.lineplot,
-                x="real_date",
-                y=val,
-                hue="xcat",
-                hue_order=xcats,
-                estimator=None,
-            )
-
-            fg.map(plt.axhline, y=0, c=".5")
-            fg.set_axis_labels("", "")
-            fg.set_titles("{col_name}")
-
-        else:
-            ax: plt.Axes = plt.gca()
-            for xc in xcats:
-                dfc: pd.DataFrame = df[df["xcat"] == xc]
-                ax.plot(dfc["real_date"], dfc[val], label=xc)
-
-            plt.axhline(y=0, c=".5")
-
-            ax.set_xlabel("")
-            ax.set_ylabel("")
-            ax.legend(
-                ncol=legend_ncol,
-                fontsize=legend_fontsize,
-            )
+            if len(xcat_labels) == len(xcats) + 1:
+                cross_mean_label = xcat_labels.pop(-1)
+
+        cdf: pd.DataFrame = df.pivot(
+            index="real_date", columns="cid", values="value"
+        ).mean(axis=1)
+        cross_mean: pd.Series = pd.Series(
```

Comment:
Why reformatting this back to numpy?

To me it looks like you already have the `cross_mean` in  `cdf` What is the difference?

Reply:
1. This file is ultimately going to be a wrapper around `msv.view.timelines`.
2. The plotting functions remain much "cleaner" if we keep them data-agnostic; so the cross-mean/other-calculations will have to happen in the `view*` wrappers.